src/components/views/elements/AddressSelector.js

```diff
@@ -124,7 +118,7 @@ module.exports = React.createClass({
                 // Saving the addressListElement so we can use it to work out, in the componentDidUpdate
                 // method, how far to scroll when using the arrow keys
                 addressList.push(
-                    <div className={classes} onClick={this.onClick(i)} onMouseEnter={this.onMouseEnter(i)} onMouseLeave={this.onMouseLeave} key={i} ref={(ref) => { this.addressListElement = ref; }} >
+                    <div className={classes} onClick={this.onClick.bind(this, i)} onMouseEnter={this.onMouseEnter.bind(this, i)} onMouseLeave={this.onMouseLeave} key={i} ref={(ref) => { this.addressListElement = ref; }} >
```

Comment:
this is fine, but for future reference:

`(i) => {this.onClick(i)}`  is preferable to `this.onClick.bind(this,i)`

A separate `AddressListEntry` component, with its own `onClick` etc methods, is more reacty and doesn't require creating new closures on each render cycle. Note that such a component doesn't even need to be in a different file.